src/data/data1d/filter1d/fft.js

```diff
@@ -32,3 +42,22 @@ export function reduce() {
     reduce: undefined,
   };
 }
+
+function generateXAxis(datum1D) {
+  const meta = datum1D.meta;
+  const baseFrequency = parseFloat(meta.$BF1);
```

Comment:
We should never use 'meta' information in any of the calculation. In spectrum1D there is an object called 'info' and the goal is to copy the interesting parameters from meta to info and have a name that means something. Please add the required parameter in the info object.

Reply:
I changed the name of getMetadata to getInfoFromMetadata as you suggested and add those parameters in info